### PR TITLE
Makes round end on wizard's death

### PR DESCRIPTION
**Pull request by carshalash**
This just really isn't a good idea, there are less admins on late at night than ever which makes wiztended common as hell whenever there are actually enough people online at these times. This change was made entirely under the idea that there will be admins on to do something if a wizzy dies early into the round which is just down right awful.

### CODE REVIEW

**Comment by kilozombie** [comment]
don't tell anyone
but I really like wiztended

**Comment by gbasood** [comment]
so wait
are you saying *actual extended* shouldnt happen without admisn on either?
🤔

**Comment by carshalash** [comment]
Welcome to metagrudge town bird boy. Population, all your pies on the floor.

**Comment by kilozombie** [comment]
>metagaming

well, it's carsh, no surprise here
:^)

also, population i'm gonna replace your cryomix with a pie mix you goddamn fucking alien

**Comment by kilozombie** [comment]
real talk though: I think the real problem is that wizard is super likely to happen on lowpop, as opposed to gamemodes which are admittedly a little more proactive and produce better rounds (traitor, double agents, even revsquad at about 25pop) which seem rarer
Everyone knows the feeling of 3 wizard rounds in a row, THAT'S when wiztended becomes a horrible horrible slog.

**Comment by carshalash** [comment]
I would very much appreciate less need of people needing to dive through logs at night due to shittery caused by these kinds of things. 

And less face it, wiztended is just plain awful. You don't get to pick spells you want to fuck around with anymore to have fun at these times, you have to worry about the rest of the crew and whether you're buttfucking them into a 3 hour round of deff from no reason to call the shuttle because you got valided 30 minutes in.

**Comment by gbasood** [comment]
>you have to worry about the rest of the crew and whether you're buttfucking them into a 3 hour round of deff from no reason to call the shuttle because you got valided 30 minutes in.

except  you don't

**Comment by carshalash** [comment]
How so?

**Comment by gbasood** [comment]
You have no responsibility as an antag to give the rest of the server a less horrible round
If you place that responsibility on yourself that's your own problem

**Comment by kilozombie** [comment]
Did you make this PR during a ragin' mages round when you thought it was wizard because the second wizard hadn't spawned yet?
...the round that just ended, I mean.

**Comment by carshalash** [comment]
I was the second wizard. The topic came up in dead chat about how single wiz is shit and inferior to ragin now so I made the PR.

**Comment by Arthurdentist** [comment]
The change wasn't made because of admins. It was made because its retarded watching players rush to kill the wizard then saying 'oh it must be ragin'!!!!!!!!!!!' when the round didn't end.

**Comment by carshalash** [comment]
That change is awful for night time when there are no admins on for hours and players are just kind of left by themselves though. And my argument is that nobody put any thought into the fact that this fucks over lowpop players.

**Comment by Arthurdentist** [comment]
Low Pop players are fucked because its low pop. Your argument could just as easily be 'lets raise the requirement on wizard rounds'. How many low pop traitor rounds end in a le ebin murderbone? quite a lot of them in my experience.

**Comment by Arthurdentist** [comment]
Also going to copy and paste my response to your previous attempt to revert this change:

"A better solution is to add more interesting and varied events.

Round ending on wizard death was always retarded and dumb and it very much lead too:

Oh shit its a wizard
Blow up the Wizard cuz round end :^) (Its okay if I blow a 9 Dev hole in the bridge because the round will end!)
Did the round end? (Y/N)
If (Y) well done Valids Secured
If (N) Oh shit its ragin' mages!!! Time to prepare for more valids" t; ArthurDentist, professional suggestion maker, Nov 15th 2016.

**Comment by kernikov** [comment]
c0nvxuew8aafmav

**Comment by Durgen** [comment]
all '-stendeds' that are not extended are shit

**Comment by kilozombie** [comment]
from the perspective of a bakerguy

round doesn't end when wizard dies, wizard dies 20 minutes in:
- round is a bit boring
- get to do lots of gimmicks and roleplay
- leave if you're bored dude

round ends when wizard dies, wizard dies 20 minutes in:
- spend 20 minutes prepping for baking, it's a long and painful process, the worst part of cooking in spess
- dumb wizard dies
- 20 minutes wasted

**Comment by eplgr** [comment]
What about looking for more lowpopmins?
What about making random events happen more often on lowpop?

**Comment by kilozombie** [comment]
A N T A G



D A A A A  A A T T UUUU   MM S

**Comment by carshalash** [comment]
Never
Ever

**Comment by gbasood** [comment]
Honestly more mins on lowpop after antags die isnt going to make your rounds less shit
People being shitters on lowpop isn't a gamemode problem, either

**Comment by Probe1** [comment]
Why do you think fresh off a permaban you are particularly equipped to make changes to the game code that are unrequested?

**Comment by carshalash** [comment]
Sometimes I wonder if you could be any more passive aggressive.

**Comment by kilozombie** [comment]
Hey now
Let's leave attacks on character for the thread, probe
Whether or not this PR is good is what's in question, not who's making the PR.

**Comment by gbasood** [comment]
That's not actually passive aggressive, that's fairly straightforward aggression.

Back to the PR: "extended-ish" rounds will not be fixed by antag datums or by fixing just wizards, and the main complaint seems to be that they're boring.
Well, that sucks, I guess you have to actually do something other than get valids.
If you'd like, I could offer up a few servers that are more deathmatch-oriented than /vg/station.

**Comment by kilozombie** [comment]
Yeah, that's the thing. I really actually yearn to have some rounds where the interest is perpetuated by the players and not just antags.
Some of my favorite rounds ever have been extended, or as we're talking about here, wiztended.

**Comment by carshalash** [comment]
My complaint is that it gives you only the option to be deathmatch and that the valid culture we have at the moment which makes it pretty difficult to be able to do anything without worrying about greymctide running at you with 2 eguns after being on the station for 5 minutes.

A vast majority of my wizard rounds I go for non lethal methods which tend to backfire due to the fact the crew can get armed out the ass fast from either armory or just the unlimited eguns from cargo. 

Saying I just want to hippie it up is a pretty big assumption desu.

**Comment by TheFriggenSawbones** [comment]
@Probe1 ? ? ? ? ?

**Comment by gbasood** [comment]
>My complaint is that it gives you only the option to be deathmatch and that the valid culture we have at the moment which makes it pretty difficult to be able to do anything without worrying about greymctide running at you with 2 eguns after being on the station for 5 minutes.

Wait, what? How does a wizard being dead mean you have to deathmatch?

As for your wizard strategies: it's not the gamemode's fault that your strats don't work

**Comment by carshalash** [comment]
Wait were you talking about post wizard death being deathmatch or wizard itself being a deathmatch? My bad man, I thought you meant something else with that.

**Comment by gbasood** [comment]
The PR is about post-wizard rounds.
I don't have any idea what you're arguing anymore, but I'm arguing that a gamemode being any form of extended due to lack of antags is fine.

**Comment by carshalash** [comment]
I'm mostly talking about the "If you'd like, I could offer up a few servers that are more deathmatch-oriented than /vg/station." thing.

**Comment by Durgen** [comment]
unban 2hotdogs

**Comment by gbasood** [comment]
@Durgen keep shitposts in thread

Since this is getting circular I'm just gonna bring attention back to one thing you said to refocus it.
>you have to worry about the rest of the crew and whether you're buttfucking them into a 3 hour round of deff from no reason to call the shuttle because you got valided 30 minutes in.

**Why is it bad that there will be a 3 hour round?**

**Comment by carshalash** [comment]
Because most of the time when rounds at these times drag on for that long, more than half of the crew go braindead because of lack of activity. People just get bored eventually from not really being able to do anything, at this time most jobs have completely their roles for at least an hour by now, sci is fully researched and have a phazon and gold slimes, engineering and mommis have had their shit done since 13:40 and medbay is begging people to injure themselves so they can have something to do. 

The only real benefit of hours dragging this long is people doing their own autism projects in places and not really everyone does that, having some form of antagonist in the round or even the threat of an antagonist will have people on their toes of some threat, knowing that it was a wizard round and that they got gunned down fast kind of removes the suspense of wondering what is going to happen next and just leaves you to do shit like autism forts. 

Autism forts are fun for people but not really everyone gets to enjoy them with their jobs

**Comment by kilozombie** [comment]
Sometimes though, people make shit interesting for various reasons. Gimmicks and roleplay keep the game alive at this point.
I don't _mind_ when admins bus in an antag, but often times things are fine without it. It just happens to be a slower round.

**Comment by Iamgoofball** [comment]
this is a confusing PR

the title implies they want the round to NOT end but they really want it to end???

**Comment by carshalash** [comment]
SHIT

**Comment by Iamgoofball** [comment]
just make it "Makes the round end when wizard dies"

**Comment by invoid9** [comment]
I agree with killerzombie, if wiz is always gonna turn into wizstended, it should have it's chance lowered to reflect the time an average round of it should take, at least on lowpop. And hey, maybe if it's rarer, more wizards will be encouraged to make the round count instead of just doing the same shit everyone is sick of. Multiple 20 minute wiz rounds where you know the round is dead is worse than 1 wizstended round where a player can at least make something of it once (if???) the wiz gets valdihunted. If the chance is lowered and this is merged, then maybe there'd still be that encouragement to make it count for a wiz. But I think even if the roundtype is rare, it's impossible to get rid of the feeling that it's a dead round once the wiz is found out (unless maybe it has a good gimmick and the crew doesnt just validhunt anyway.)

Besides, if the problem is solely during lowpop, then the solution should only affect lowpop. This solution affects every pop and drastically worsens highpop.

**Comment by carshalash** [comment]
I have pretty negative opinions about single wizzy at high pop anyway but people seem to really like summon guns/magic

**Comment by ghost** [comment]
Solo wizard makes them take summon guns/magic since dying makes it into extended and it's best to have some possible antags in case you die.
The best solo wiz build is summon magic x5 tbh

**Comment by Shadowmech88** [comment]
The problem with wiztended is that extended works best when players don't *know* it's extended.

Once you kill the wizard, you can reasonably assume that all the antags are gone, which means whoever is left will either bury themselves in autism or commit suicide, because they *know* that they are safe and can work until the crew requests a transfer.

**Comment by SonixApache** [comment]
code wiz to turn into mixed/autotot/morewiz after a random time of just lolrevert
nobody likes 5 minute fireballwiz rounds

**Comment by kilozombie** [comment]
>nobody likes 5 minute fireballwiz rounds

but
but have you been _reading_ this _thread_ though
I don't mind 5 minute fireballwiz rounds so long as the rest of the round has some stuff going on. And that stuff doesn't have to be antag-related.
It's only unpleasant, for me, if it happens a couple times in a row.

**Comment by ghost** [comment]
How about replace solo wiz with wizard+changeling like we have wiz+vampire
Maybe we'll get to see some memes happen with lings and magic

**Comment by mtpowell** [comment]
I'd like to see this, every experience I've had with a Wiz dying early has admins bussing in more Wizs until the round is over.

**Comment by JustSumBody** [comment]
>like we have wiz+vampire

This isn't a thing, that was most likely forced mixed.

**Comment by Durgen** [comment]
maybe we should remove solo wizard and buff the chances of ragin

**Comment by kilozombie** [comment]
I'd like to see solo wiz just be made more unlikely.
Ragin' is HEAVILY anti-comfy.

**Comment by FudgePucker** [comment]
Oh man I hate early round end on wizard death.
Nothing is worse than wizard rounds back to back, don't even bother setting up anything because halfway though the round ends. I have made over three lengthy posts about why round end wizard deaths are bad.

Reduce the chances of solo wizard rounds, increase the amount of admins, or set up a system to spawn more antags if all antags are dead or being fucking lame. Please do not bring back the hell that is instant end wizard rounds.

**Comment by TheFriggenSawbones** [comment]
@Iamgoofball Hey, I think you're on the wrong github again.
https://github.com/tgstation/tgstation is where you belong, I hope this helps!

**Comment by yclatious** [comment]
What, you don't enjoy extended after a antag round? What are you, a fag or something, learn to interact with the crew and indulge your own autism in interesting ways, remake the ghetto bar, steal like the best nigger to ever be and give it all back to the right people, chill at the bar, run the bar, chill with sec, steal sec, indulge your inner kleptomaniac and steal everything like you are a bipedal space chicken, hack a safe only to find shit to trade the vox for, make weed booze and give it to every greyshit you can find, run ghetto botany for comfy points, pray to the adminbus after you shove Ian in a washing machine, play the arcade and fuck around enough to get a admin to give you a hacked jukebox from your shitposting efforts at decorating the bar with your toys, actually trade and chill with the vox, make skeleton puns near skelememes, ask a plasmeme to photocopy his own ass and sent it to centcom, have some fun that isn't you making a sprite horizontal, god damn.

We play in a game where a clown can set up a fucking thermonuclear gas heating engine, where a space wizard riding a car can turn everyone into a clown, where changelings from the thing can turn everyone into mimes and have eaten recently deceased people and impersonated them at their own funeral, where you can fully fuck up highly trained nuclear operatives with just a banana peel, bolas made out of a dog, cat and screwdriver alongside a toolbox and you want to spend the time you have playing this game in getting valids, just toolboxing people who are there to make the interesting parts happen?

Cmon carshashalala, you are better than this. Dont give in to hippie, don't give in to bay, we preserve balance and rounds continuing after wizards go the way of the shotgun shell to the head is fine by me and many others. Maybe you should try and have some fun like we do, its fun to do some heavy roleplay once in a blue moon.

**Comment by Iamgoofball** [comment]
many thanks and upvotes my friend :) i thought this was the reddit station github

**Comment by Yelier** [comment]
What purpose would this serve besides wasting the time of the players that aren't the wiz/actively hunting the wiz? Making sure that you can end the round quickly because you didn't get your antag roll? A significant portion of the jobs require time to be satisfying. Why would I bother even attempting to build something cool as engineer when I know the round's going to end in 10 minutes when everyone finds out that it's wizard? The same thing applies to a large portion of the jobs. I'm not saying that every round should be 3 hours long and have half of the server braindead from utter boredom, but I'd much rather have a fun and fulfilling round that went on too long than a 15 minute long one that was neither fun NOR fulfilling.

**Comment by SkowronX** [comment]
downvote this muthafucka

**Comment by Megamaw** [comment]
while i think ending the round upon the wizard's death encourages validhunting and ultimately makes a far less interesting game for the wizard, and thusly the crew (why bother figuring stuff out when you can just blow up the station, kill everyone, and reroll for antag?), there are a few underlying issues that make any extended round a nightmare.

@gbasood there are two main reasons a long extended round is 'bad'.

the first reason is without antags, people get bored. but this isn't the main reason. in fact, with people that aren't antags but just kinda jerks, there's still plenty of fun to be had. there's a bigger underlying reason i've very seldom seen addressed.

job slots. midway through the round, you can probably only pick like, botanist, officer, medical doctor, and assistant. a lot of the time, people don't want to play these roles! I know personally i've just waited for round-end when it's like that b/c hell, I'm not here to play assistant. and as people leave the game as time goes on, there's a bunch of crew members in fun jobs who went braindead.

so without new people entering the round (due to all the more desirable jobslots being taken), and with a dwindling current staff, even the usual shenanigans start slowing to a crawl as the station starts running low on the roundstart resources. people get bored, go home, not very satisfying.

so while people say the tensioner and antag datums are the end-all be-all for this sort of thing, it really doesn't do shit if there's no one new entering the round, because no matter what you do, people will leave the game mid round. and over 3 hours, so many people have left that there's not much interesting you can do anymore.


my proposal to fix this is some sort of way to increase job caps over time. janitor went braindead at roundstart? i've seen other servers where HoPs can simply alter the job join caps, so maybe they could order a new janitor, and someone else could play!

idk. but that's what i've experienced. 

tl;dr
the less job options you have, the more likely you are to wait for round-end. the less people willing to join the round, the more braindead people begin to outnumber living players, and the braindead issue snowballs from there. we need to fix the whole 'you only ever get ONE janitor/HoS/mime/etc' problem, because that's what hurts round longevity.

**Comment by nuklearcellphoneg** [comment]
THE ROUND HAS ENDED (12:06)

THE WIZARD'S SPELLS WERE:

FIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALLFIREBALL

**Comment by carshalash** [comment]
Thats just sad. Guy gets multiple spellbooks and all he does is get more fireballs

**Comment by nuklearcellphoneg** [comment]
>round ends
>takes five minutes
>new round begins
>takes about five minutes
>round starts
>wiz kills itself on accident
>round ends
>five minutes to start up
>round begins
>wiz again, successfully does not kill itself
>oh the crew killed it
>five minutes to end
>five minutes to start up again
>2 players have readied up

**Comment by Duny-** [comment]
Muh antag reroll bwaaaah
At least mindswap is actually a spell you can use now, and wizard rounds aren't just validhunters running after the wizard while the rest suicide succumbs because no matter what they do (try to kill the wizard or not) the round will end soon 99% of the time.

**Comment by Iamgoofball** [comment]
Maybe you should consider moving startup into subsystems for faster boots

**Comment by nuklearcellphoneg** [comment]
I was meming a bit, but wiz ending on death is pretty annoying- I find blob acceptable, more because it winds up with most of the crew pooling resources to overcome the blob, which tends to go hand in hand with the peoples' autism projects. 

Wiz dead, round over, is pretty lame if you ask me. You could go the whole round without any notion that a wiz exists, and poof, round over.

**Comment by Probe1** [comment]
ijxhzcunccjxi

**Comment by ihadtoregisterforthis** [comment]
The problem is not the round becoming extended and lasting 3 hours, the problem is the players knowing that it will invariably be extended and there's no way to stop it from lasting 3 hours. There's no reason to call the shuttle, there are no problems to solve, the activities available run out. The wizard can become something not more relevant than a carp entering through escape. The round is already over and for many all there's left is wait for the captain to stub their toe or the server to crash.
Natural extended rounds somehow work out and stuff actually happens, but the thrill of the danger at any second is always there. Wizards don't stir the station for more time than their lifespan.

**Comment by Shadowmech88** [comment]
I agree.
It wouldn't be such a problem if we had the tensioner for random events, so that the round itself could threaten the crew after the wizard dies.
Maybe once antag datums are finished we can make it turn into a different type of round after the wizard dies.
